test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightDeltaWriterTests.cs

```diff
@@ -2119,7 +2119,7 @@ public void WriteNestedDeltaResourceSetIn40ShouldFail(bool isResponse)
 
         [InlineData(/*isResponse*/true)]
         [InlineData(/*isResponse*/false)]
-        [Theory]
+        [Theory(Skip ="Should not throw exception")]

```

Comment:
Looking at the number of tests you are allowing to pass in this PR confirms my fears that we should not be doing this fix.
From the spec, `Deep insert` was introduced in `v4` and Deep update in `v4.01`. See https://docs.oasis-open.org/odata/new-in-odata/v4.01/cn04/new-in-odata-v4.01-cn04.html#sec_NewDeepUpdate

Therefore we should leave the `OData-Version` validation as it is and ensure properly document that `Deep update` only works in `v4.01`.

What do you think? @mikepizzo @xuzhg